src/api/routers/topic.ts

```diff
@@ -22,14 +22,23 @@ export const topicRouter = router({
     )
     .query(async (opts) => {
       const isCreator = opts.input.username === opts.ctx.user?.username;
+      const normalizedTitle = normalizeTitle(decodeURIComponent(opts.input.title));
```

Comment:
Did you find it necessary to decode? It's my understanding that TRPC automatically encodes on the client-side and & decodes on the server side for us ([specification which seems to imply this](https://trpc.io/docs/rpc#methods---type-mapping)).
```suggestion
      const normalizedTitle = normalizeTitle(opts.input.title);
```